### DISCUSSION TITLE
Custom node as Handle component

### DISCUSSION THREAD

**Discussion by protzman** [xyflow/xyflow#1097] [New Features]
It would be helpful if we could either wrap our custom nodes within a `<Handle />` component or the handle component itself could have `position={Position.self}` or something which would 'center' the handle as the component, although maybe make the handle invisible. This would allow _n_ children nodes to flow in all directions instead of just the four. Say I have 5 children nodes - I don't necessarily want all to come out of the bottom as that increases the x axis of my whole diagram. While I can zoom out and pan, It would be beneficial to be able to write a function that could calculate the layout based on 360 degrees, and just place my nodes all from a center point versus the bottom or left, or choosing to split up like 3 from the bottom handle, one from left handle and one from right handle.

**Comment by likelion** [comment]
Ideally, there should be a way to make "handleless" nodes, where the whole node is both the source and the target. Edge creation can start when you double click the node (or programmatically based on some event in the node) and finish when you drop it anywhere in the node. A virtual edge endpoint position can reside in the centre of the node, and the edge end can flow along the side of the node as the node is moved. One can also combine nodes with and without handles in the same flow: so, for example, an edge can originate in the handle and terminate in the node. Rendering can be done using custom edges but the behaviour, I guess, needs to be supported by the lib.
![Screenshot 2021-06-03 at 13 40 34](https://user-images.githubusercontent.com/3233957/120639152-587f0500-c471-11eb-97bb-001430638015.png)
